src/ipred.rs

```diff
@@ -739,38 +739,38 @@ fn get_filter_strength(wh: c_int, angle: c_int, is_sm: bool) -> c_int {
 }
 
 #[inline(never)]
-unsafe fn filter_edge<BD: BitDepth>(
-    out: *mut BD::Pixel,
+fn filter_edge<BD: BitDepth>(
+    out: &mut [BD::Pixel],
+    out_off: usize,
```

Comment:
If we're not indexing backwards (we're always doing `out_off + _ as usize`, unless that's a mistake), why can't we just slice `out` and not have a separate `out_off` arg?

Reply:
Ah you're right we can do that. I initially thought we were doing negative indexing on `out`, but I later realized that wasn't the case and didn't think to remove `out_off`. Changed.